qemu/tests/migration_multi_host_with_kdump.py

```diff
@@ -0,0 +1,184 @@
+from autotest.client.shared import error
+from autotest.client.shared.syncdata import SyncData
+from virttest import env_process
+from virttest import utils_test
+from generic.tests import kdump
+
+
+@error.context_aware
+def run(test, params, env):
+
+    """
+    KVM multi-host migration ping pong test:
+
+    Migration execution progress is described in documentation
+    for migrate method in class MultihostMigration.
+
+    The test procedure:
+    1) starts vm on master host.
+    2) Configure kdump inside guest
+    3) Implement kdump inside guest and pin the task to 1 cpu
+    4) do ping pong migration until guest was verified to reboot succeed
+    5) check "/var/crash" directory inside guest, whether dump file generates
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    mig_protocol = params.get("mig_protocol", "tcp")
+    base_class = utils_test.qemu.MultihostMigration
+    if mig_protocol == "fd":
+        base_class = utils_test.qemu.MultihostMigrationFd
+    if mig_protocol == "exec":
+        base_class = utils_test.qemu.MultihostMigrationExec
+    if "rdma" in mig_protocol:
+        base_class = utils_test.qemu.MultihostMigrationRdma
+
+    class TestMultihostMigration(base_class):
+
+        def __init__(self, test, params, env):
+
+            super(TestMultihostMigration, self).__init__(test, params, env)
+            self.srchost = self.params.get("hosts")[0]
+            self.dsthost = self.params.get("hosts")[1]
+            self.vms = params.objects("vms")
+            self.vm = self.vms[0]
+            self.is_src = params["hostid"] == self.srchost
+            self.stop_migrate = False
+            self.migration_timeout = int(params.get("migration_timeout",
+                                                    "1500"))
+            self.crash_timeout = float(params.get("crash_timeout", 360))
+            self.def_kernel_param_cmd = ("grubby --update-kernel=`grubby"
+                                         " --default-kernel`"
+                                         " --args=crashkernel=128M@16M")
+            self.kernel_param_cmd = params.get("kernel_param_cmd",
+                                               self.def_kernel_param_cmd)
+            def_kdump_enable_cmd = ("chkconfig kdump on &&"
+                                    " service kdump restart")
+            self.kdump_enable_cmd = params.get("kdump_enable_cmd",
+                                               def_kdump_enable_cmd)
+            def_crash_kernel_prob_cmd = ("grep -q 1 /sys/kernel/"
+                                         "kexec_crash_loaded")
+            self.crash_kernel_prob_cmd = params.get("crash_kernel_prob_cmd",
+                                                    def_crash_kernel_prob_cmd)
+            self.crash_cmd = params.get("crash_cmd",
+                                        "echo c > /proc/sysrq-trigger")
```

Comment:
Confirmed with kdump owner, for rhel, sysrq function are enabled as default,
but for fedora, it's not, it's better add it.